api/nodes/serializers.py

```diff
@@ -164,6 +164,11 @@ class NodeSerializer(JSONAPISerializer):
         related_view_kwargs={'node_id': '<forked_from_id>'}
     )
 
+    templated_from = RelationshipField(
+        related_view=lambda n: 'registrations:registration-detail' if getattr(n, 'template_node', None) else 'nodes:node-detail',
+        related_view_kwargs={'node_id': '<templated_from_id>'}
```

Comment:
The gymnastics with the model are probably unnecessary. You can probably just do `'node_id': <'template_node._id>`